guardian/dotcom-rendering
# Dblatcher/stand alone pages

## What does this change?

 - introduces a the `FENewslettersPageType`, representing a request from frontend to render the Newsletters page and the `DCRNewslettersPageType` representing the validated and parsed data for rendering the page in DCR
 - Adds the validation functions and schema for the new types.
 - Adds the "/EmailNewsletters" route for validating, parsing and rendering a newsletters  page in response to a request from frontend for the newsletters page
 - Adds the "/EmailNewslettersTestData" route (dev-server only) to render a newsletters  page using static data (to allow faster local dev work on the page component without running frontend)
 - creates a set of components to:
   - render the standard DCR page template, with a  Header and Footer, using the data from the `DCRNewslettersPageType`
   - render  placeholder page content using the newsletter data received

## Why?

See https://github.com/guardian/frontend/pull/25876 - We have a new design of the newsletters page to implement in DCR. Adding this option for DCR to render a placeholder version of the page while the design is implemented will be the first step replacing the frontend-rendered version.

note - the page rendered for this PR is only a placeholder, not the new design.

## Testing

local test page:
 - run DCR locally
 - open "http://localhost:3030/EmailNewslettersTestData"

frontend render
 - run DCR locally
 - run Frontend locally, using the branch at https://github.com/guardian/frontend/pull/25876 
 - requesting "http://localhost:9000/email-newsletters?dcr=true" will render the DCR version of the page (there may be timeout error the first time as frontend will be compiling on demand - should render correctly after reloading)
 - requesting "http://localhost:9000/email-newsletters" will render the current frontend version of the page as normal

## Screenshots

<img width="1254" alt="Screenshot 2023-02-06 at 12 38 05" src="https://user-images.githubusercontent.com/30567854/216981153-d417a019-8350-45c6-8550-3ac0ed0e78c3.png">


-->
